libs/SmartSync/SmartSync/Classes/Target/SFParentChildrenSyncUpTarget.m

```diff
@@ -0,0 +1,591 @@
+}
+
+
+- (SFRestRequest *)buildRequestForParentRecord:(NSDictionary *)record fieldlist:(NSArray *)fieldlist {
+    return [self buildRequestForRecord:record fieldlist:fieldlist isParent:true useParentIdReference:false parentId:nil];
+}
+
+- (SFRestRequest *)buildRequestForChildRecord:(NSDictionary *)record
+                         useParentIdReference:(BOOL)useParentIdReference
+                                     parentId:(NSString *)parentId {
+
+    return [self buildRequestForRecord:record fieldlist:nil isParent:false useParentIdReference:useParentIdReference parentId:parentId];
+}
+
+- (SFRestRequest *)buildRequestForRecord:(NSDictionary *)record
+                               fieldlist:(NSArray *)fieldlist
+                                isParent:(BOOL)isParent
+                    useParentIdReference:(BOOL)useParentIdReference
+                                parentId:(NSString *)parentId {
+
+    if (![self isDirty:record]) {
+        return nil; // nothing to do
+    }
+
+    SFParentInfo *info = isParent ? self.parentInfo : self.childrenInfo;
+    NSString *id = record[info.idFieldName];
+
+    // Delete case
+    BOOL isCreate = [self isLocallyCreated:record];
+    BOOL isDelete = [self isLocallyDeleted:record];
+
+    if (isDelete) {
+        if (isCreate) {
+            return nil; // no need to go to server
+        } else {
+            return [[SFRestAPI sharedInstance] requestForDeleteWithObjectType:info.sobjectType objectId:id];
+        }
+    }
+        // Create/update cases
+    else {
+        fieldlist = isParent
+                ? isCreate
+                        ? (self.createFieldlist ? self.createFieldlist : fieldlist)
+                        : (self.updateFieldlist ? self.updateFieldlist : fieldlist)
+                : isCreate
+                        ? self.childrenCreateFieldlist
+                        : self.childrenUpdateFieldlist;
+
+        NSMutableDictionary *fields = [self buildFieldsMap:record fieldlist:fieldlist idFieldName:info.idFieldName modificationDateFieldName:info.modificationDateFieldName];
+        if (parentId) {
+            fields[((SFChildrenInfo *) info).parentIdFieldName] = useParentIdReference ? [NSString stringWithFormat:@"@{%@.%@}", parentId, @"id"] : parentId;
+        }
+
+        if (isCreate) {
+            return [[SFRestAPI sharedInstance] requestForCreateWithObjectType:info.sobjectType fields:fields];
+        } else {
+            return [[SFRestAPI sharedInstance] requestForUpdateWithObjectType:info.sobjectType objectId:id fields:fields];
+        }
+    }
+}
+
+- (NSDictionary *)parseIdsFromResponse:(NSDictionary *)refIdToResponses {
+    NSMutableDictionary *refIdToId = [NSMutableDictionary new];
+    for (NSString *refId in [refIdToResponses allKeys]) {
+        NSDictionary *response = refIdToResponses[refId];
+        if ([((NSNumber *) response[@"httpStatusCode"]) unsignedIntegerValue] == 201) {
+            NSString *serverId = response[@"body"][@"id"];
```

Comment:
Actually no.
With standard/custom sobjects, the identity field when you get objects or query them with soql is "Id" (capital i). However when you create one sobject, the identity field in the response is "id" (small case i) - see https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_sobject_create.htm.

We don't have a way to specify an alternative identity field for the create response. If they have one, they will need to override the createOnServer method. Since alternate identity fields are only useful if you use other end points for your CRUD operations, you should be overriding that method anyway.